github-readme.php

```diff
@@ -13,38 +13,111 @@
-	
-	$transient="github_readme_" . $repo . "_" . $trim;
-	if ( false === ( $html = get_transient($transient))) 
-	{
-	 	$url="https://api.github.com/repos/" . $repo . "/readme";
-	
-		$ch = curl_init();
-		$timeout = 5;
-		curl_setopt($ch, CURLOPT_URL, $url);
-		curl_setopt($ch,CURLOPT_USERAGENT,'WordPress');
-		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-		curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, $timeout);
-		$data = curl_exec($ch);
-		curl_close($ch);
-
-		$json=json_decode($data);
- 		$markdown=base64_decode($json->content);
-		if ($trim>0)
-		{
-			$markdown = implode("\n", array_slice(explode("\n", $markdown), $trim)); 
-		}
-	
-		$html = Markdown::defaultTransform($markdown);
-	 	set_transient($transient, $html , 12 * HOUR_IN_SECONDS);
-	}
-	return $html;	
+    extract( shortcode_atts( array(
+        'repo' => 'octalmage/GitHub Shortcode',
+        'trim' => 0
+    ), $atts ) );
+
+    $transient="github_readme_" . $repo . "_" . $trim;
+    if ( false === ( $html = get_transient($transient)))
+    {
+        $url="https://api.github.com/repos/" . $repo . "/readme";
+
+        $ch = curl_init();
+        $timeout = 5;
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch,CURLOPT_USERAGENT,'WordPress');
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, $timeout);
+        $data = curl_exec($ch);
+        curl_close($ch);
+
+        $json=json_decode($data);
+        $markdown=base64_decode($json->content);
+        if ($trim>0)
+        {
+            $markdown = implode("\n", array_slice(explode("\n", $markdown), $trim));
+        }
+
+        $html = Markdown::defaultTransform($markdown);
+        set_transient($transient, $html , 12 * HOUR_IN_SECONDS);
+    }
+    return $html;
 }
+
+function github_markdown_func( $atts )
+{
+    extract( shortcode_atts( array(
+        'repo' => 'octalmage/GitHub Shortcode',
+        'trim' => 0,
+        'cache' => 60,
+        'file' => '/readme',
+        'branch' => 'master'
+    ), $atts ) );
+
+    $transient="github_markdown_" . $repo . "_" . $file . "_" . $trim;
+    if ( false === ( $html = get_transient($transient)))
+    {
+        $url="https://raw.githubusercontent.com/" . $repo . "/" . $branch . "/" . $file;
+
+        $ch = curl_init();
+        $timeout = 5;
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch,CURLOPT_USERAGENT,'WordPress');
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, $timeout);
+        $markdown = curl_exec($ch);
+        curl_close($ch);
+
+        if ($trim>0)
+        {
+            $markdown = implode("\n", array_slice(explode("\n", $markdown), $trim));
+        }
+
+        $html = Markdown::defaultTransform($markdown);
+        set_transient($transient, $html, $cache);
+    }
+    return $html;
+}
+
+function github_wikipage_func( $atts )
+{
+    extract( shortcode_atts( array(
+        'user' => '',
```

Comment:
Hmm why did you use `user` and `project` instead of repo like the other shortcodes?